openxr/src/secondary_view.rs

```diff
@@ -0,0 +1,12 @@
+use crate::*;
+
+pub struct SecondaryViewState {
+    pub ty: ViewConfigurationType,
+    pub active: bool,
+}
+
+pub struct SecondaryEndInfo<'a, 'b, 'c, G: Graphics> {
+    pub ty: ViewConfigurationType,
+    pub environment_blend_mode: EnvironmentBlendMode,
+    pub layers: &'a [&'b CompositionLayerBase<'c, G>],
+}
```

Comment:
What's the benefit of using three different lifetimes here, given that we bind them all to the same lifetime in `end_secondary` anyway?

Reply:
It's not bound to the same lifetime, that syntax is just explicitly requesting they be elided.

Reply:
Isn't that exactly what lifetime elision does, though? Bind everything to a single lifetime?

Reply:
ping?

Reply:
No, it does not, lifetime elision binds input and output lifetimes to the same lifetime but multiple input lifetimes get elided to different lifetimes

Reply:
Right you are. Can you clarify why using multiple lifetimes here is useful? What's an example of a call that it permits which would be impossible if only a single lifetime was used?

Reply:
Uh, _any_ call? If you have a nested lifetime it's not typical that the outer and inner lifetime are the same. How can they be, the intermediate object (the slice here) would _have_ to have been constructed after its contents were referenceable.

What _usually_ happens is that Rust lifetimes are, in many cases, squishy, and when you have an API with equal lifetimes things will usually be squished to fit. But this involves [lifetime variance](https://doc.rust-lang.org/nomicon/subtyping.html) and it's best to just not rely on it.

Like I said, the default behavior of lifetime elision is to introduce multiple lifetimes anyway, multiple lifetiems are more common than you'd expect in rust, they're just usually elided (and can't be in structs).

Reply:
Right, I'm asking if there's a specific variance issue this solves, since normally subtyping is plenty to account for the other issues you describe.

Reply:
Probably not, but it could in the future. I don't see any harm to do this here, this is an init struct that you initialize and immediately pass in

Reply:
Fair enough.